lib/roo_on_rails/logger.rb

```diff
@@ -0,0 +1,92 @@
+require 'logger'
+require 'delegate'
+require 'roo_on_rails/logfmt'
+
+module RooOnRails
+  class Logger < SimpleDelegator
+    def initialize(io = STDOUT)
+      @show_timestamp = io.tty?
+      logger = ::Logger.new(io).tap do |l|
+        l.formatter = method(:_formatter)
+      end
+      super(logger)
+      set_log_level
+    end
+
+    def with(context = {})
+      unless block_given?
+        return Proxy.new(self, context)
+      end
+
+      new_context = (_context_stack.last || {}).merge(context)
+      Thread.handle_interrupt(Exception => :never) do
+        begin
+          _context_stack.push(new_context)
+          Thread.handle_interrupt(Exception => :immediate) do
+            yield self
+          end
+        ensure
+          _context_stack.pop
+        end
+      end
+      nil
+    end
+
+    def clear_context!
+      _context_stack.replace([{}])
+    end
+
+    def set_log_level
+      self.level = ::Logger::Severity.const_get(ENV.fetch('LOG_LEVEL', 'DEBUG'))
+    end
+
+    private
+
+    class Proxy < SimpleDelegator
+      def initialize(logger, context)
+        @context = context
+        super(logger)
+      end
+
+      %w[add debug info warn error fatal unknown].each do |name|
```

Comment:
Do you think that small savings in avoided boilerplate really warrant those arcane incantations?

Reply:
Er, I'm not sure what's arcane here? The usage of `define_method`? This is actually _less_ arcane than [the way ActiveSupport does it](https://github.com/rails/rails/blob/master/activesupport/lib/active_support/logger.rb#L92-L98), in my humble opinion.

Reply:
That, in conjunction with `__getobj__` and `public_send`. And it saves you maybe 12 lines of boilerplate. I'm not going to block on that or anything, but I hate metaprogramming, and IMO this is one of those cases where the benefit is not worth the cost.